code/computational_geometry/src/cohen_sutherland_lineclip/Cohen-SutherlandLineClippingAlgorithm.cpp

```diff
@@ -0,0 +1,153 @@
+                    break;
+                }
+                else if (code1 & code2)
+                {
+                    // If both endpoints are outside rectangle,in same region.
+                    break;
+                }
+                else
+                {
+                    // Some segment of line lies within the rectangle.
+                    int code_out;
+                    double x, y;
+                    // At least one endpoint lies outside the rectangle, pick it.
+                    if (code1 != 0)
+                        code_out = code1;
+                    else
+                        code_out = code2;
+                     /*
+                      * Find intersection point by using formulae :
+                      y = y1 + slope * (x - x1)
+                      x = x1 + (1 / slope) * (y - y1)
+                      */
+                    if (code_out & TOP)
+                    {
+                        // point is above the clip rectangle
+                        x = x1 + (x2 - x1) * (y_max - y1) / (y2 - y1);
+                        y = y_max;
+                    }
+                    else if (code_out & BOTTOM)
+                    {
+                        // point is below the rectangle
+                        x = x1 + (x2 - x1) * (y_min - y1) / (y2 - y1);
+                        y = y_min;
+                    }
+                    else if (code_out & RIGHT)
+                    {
+                        // point is to the right of rectangle
+                        y = y1 + (y2 - y1) * (x_max - x1) / (x2 - x1);
+                        x = x_max;
+                    }
+                    else if (code_out & LEFT)
+                    {
+                        // point is to the left of rectangle
+                        y = y1 + (y2 - y1) * (x_min - x1) / (x2 - x1);
+                        x = x_min;
+                    }
+                    // Intersection point x,y is found.
+                    // Replace point outside rectangle by intersection point.
+                    if (code_out == code1)
+                    {
+                        x1 = x;
+                        y1 = y;
+                        code1 = generateCode(x1, y1);
+                    }
+                    else
+                    {
+                        x2 = x;
+                        y2 = y;
+                        code2 = generateCode(x2, y2);
+                    }
+                }
+            }
+            if (accept)
+            {
+                std::cout <<"Line accepted from " <<"("<< x1 << ", "
+                     << y1 << ")" <<  " to "<< "(" << x2 << ", " << y2 << ")" << std::endl;
+            }
+            else
+                std::cout << "Line rejected" << std::endl;
+        }
+        int main()
+        {
+            CohenSutherLandAlgo c;
+            c.getCoordinates();
+            c.getClippingRectangle();
+            c.cohenSutherland();
+            return 0;
```

Comment:
```suggestion
```

Reply:
Sorry, I can't get this change?

Reply:
@piyush26c The line is unnecessary because `return 0` in C++ is implicit.

Reply:
Yeah, I got it...